neqo-transport/src/connection/mod.rs

```diff
@@ -2547,10 +2547,10 @@ impl Connection {
         max_datagrams: NonZeroUsize,
     ) -> Res<SendOptionBatch> {
         let packet_tos = path.borrow().tos();
-        let mut send_buffer = Vec::new();
+        let mtu = path.borrow().plpmtu();
+        let mut send_buffer = Vec::with_capacity(mtu);
         let mut max_datagram_size = None;
         let mut num_datagrams = 0;
-        let mtu = path.borrow().plpmtu();
```

Comment:
Note I tried this in the past (https://github.com/mozilla/neqo/pull/2782), but wasn't able to get a clear signal that this is an actual improvement. E.g. for small UDP datagrams (ACKs only), it hurt more than it helps.

If we do get a signal here, great, let's do this.